lib/api.js

```diff
@@ -0,0 +1,113 @@
+import base from 'taskcluster-base';
+import common from './common';
+import storage from './storage';
+import _ from 'lodash';
+
+/**
+ * Check to see if a resource an expiration has passed....
+ **/
+function hasExpired(expires) {
+  let currentTime = new Date();
+  let expireTime = new Date(expires);
+  if (currentTime > expireTime) {
+    return true;
+  }
+  return false;
+};
+
+/** API end-point for version v1/
+ *
+ * In this API implementation we shall assume the following context:
+ * {
+ *   publisher:      // publisher from base.Exchanges
+ * }
+ */
+var api = new base.API({
+  title:        "TaskCluster Secrets API Documentation",
+  description: [
+    "The secrets service, typically available at",
+    "`tools.taskcluster.net`, is responsible for managing",
+    "secure data via TaskCluster scopes."
+  ].join('\n')
+});
+
+// Export API
+module.exports = api;
+
+/** Define tasks */
+api.declare({
+  method:      'put',
+  route:       '/scope/:secretScope',
+  name:        'secretWrite',
+  input:       common.SCHEMA_PREFIX_CONST + "secret.json#",
+  scopes:      ['secrets:write'],
+  title:       'Create New Secret',
+  description: 'Set a secret associated with some scope.'
+}, async function(req, res) {
+    let {secretScope} = req.params;
+    let params = req.body;
+    req.satisfies([secretScope])
+    if (params.expires) {
+      let expires = new Date(params.expires);
+      if (expires == 'Invalid Date') {
+        res.status(400).send('expires: must be a datetime (YYYY-MM-DD hh:mm:ss)');
+        return;
+      }
+    }
+    this.storageObject.write(secretScope, params);
```

Comment:
hmm... good question... I thought we should have create and a separate modify... But yeah, maybe we don't need it...

Just `secret.set(name, secret)` and `secrets.get(name)`.